docs/src/reference/workflow-instance-creation.md

```diff
@@ -0,0 +1,99 @@
+There are two commands to create a workflow instance.
+
+### Create and Execute Asynchronously
+
+ A workflow that has a [none start event](/bpmn-workflows/none-events/none-events.html#none-start-events) can be started explicitly using the command [CreateWorkflowInstance](/reference/grpc.html#createworkflowinstance-rpc).
+ When the broker receives this commands, it creates a new workflow instance and immediately respond with the workflow instance id.
+ The execution of the workflow happens after the response is send.
+
+ ![create-workflow](/reference/workflow-instance-creation/create-workflow.png)
+
+ <details>
+   <summary>Code example</summary>
+   <p>Create a workflow instance:
+
+   ```
+   zbctl create instance "order-process" --insecure
+   ```
+   Response:
+   ```
+   {
+     "workflowKey": 2251799813685249,
+     "bpmnProcessId": "order-process",
+     "version": 1,
+     "workflowInstanceKey": 2251799813686019
+   }
+
+   ```
+
+   </p>
+ </details>
+
+### Create and Await Results
+
+Typically, workflow creation and execution are decoupled.
+However, there are use-cases that need to collect the results of a workflow when it's execution is completed.
+The [CreateWorkflowInstanceWithResult](/reference/grpc.html#createworkflowinstancewithresult-rpc) command allows you to “synchronously” execute workflows and receive the results via a set of variables.
+The response is send when the workflow execution is completed.
+
+ ![create-workflow](/reference/workflow-instance-creation/create-workflow-with-result.png)
+
+Failure scenarios that are applicable to other commands are applicable to this command. Clients may not get a response in the following cases even if the workflow execution is completed successfully.
+- Leader failover: When the broker that is processing this workflow crashed, another broker continues the processing. But it does not send the response because the request is registered on the other broker.
+- Gateway failure: If the gateway to which the client is connected failed, broker cannot send the response to the client.
+- gRPC timeout: If the gRPC deadlines are not configured for long request timeout, the connection may be closed before the workflow is completed.
+
+This command is typically useful for short running workflows and workflows that collect information.
+If the workflow mutates system state, or further operations rely on the workflow outcome response to the client, take care to consider and design your system for failure states and retries.
+Note that, when the client resend the command, it creates a new workflow instance.
+
+<details>
+  <summary>Code example</summary>
+  <p>Create a workflow instance and await results:
+
+```
+./zbctl create instance benchmark --insecure --withResult
+```
+Response:
+```
+{
+  "workflowKey": 2251799813685249,
+  "bpmnProcessId": "benchmark",
+  "version": 1,
+  "workflowInstanceKey": 2251799813686045,
+  "variables": "{}"
```

Comment:
I think, in the example, it would make sense to return some variables.
Why should we wait for the result if it contains no variables? :sweat_smile: